### DISCUSSION TITLE
Custom page title when running server from docker container

### DISCUSSION THREAD

**Discussion by aranvir** [h2oai/wave#2030] [Q&A] [answered]
Hi, I'm running my application as a docker container, mostly for some docker compose convenience with other apps.

To give the app a more custom style I would like to change the title and maybe the page icon (also to make it easier to distinguish different apps from another). I know that I can just edit the `index.html` file but for building a docker image I was running into some problems:

First I thought I'll just overwrite the `index.html` with a custom copy but that did not work out... as I found out the filename of `"./wave-static/index-98d98a8e.js"` is not static and my local index file pointed to a differently named index.js file than my docker image.

So my solution now is a bit awkward since I just use `RUN sed -i 's/<title>H2O Wave/<title>My cool app/' /usr/local/www/index.html` to customize the index file.

I'm not sure if there are better solutions for this. If possible, it would be nice if stuff like this could be customized via an environment variable (title, description, icon).

**Comment by mturoci** [comment, marked as answer]
Sure, all you need is a `meta_card` that allows you to set both title and icon:

https://wave.h2o.ai/docs/examples/meta-title/
https://wave.h2o.ai/docs/examples/meta-icon/

The icon should work on all browsers except Safari which doesn't support dynamic icons for some reason. The only workaround for this is to change the index.html as you described above.

**Reply by aranvir** [reply]
That makes so much more sense than my solution........ thanks!!
